cmd/config-migrate.go

```diff
@@ -2409,29 +2410,32 @@ func migrateV27ToV28() error {
 	return nil
 }
 
-// Migrates '.minio.sys/config.json' v27 to v28.
+// Migrates '.minio.sys/config.json' v28 to v29.
 func migrateMinioSysConfig(objAPI ObjectLayer) error {
-	return migrateV27ToV28MinioSys(objAPI)
+	return migrateV28ToV29MinioSys(objAPI)
 }
 
-func migrateV27ToV28MinioSys(objAPI ObjectLayer) error {
+func migrateV28ToV29MinioSys(objAPI ObjectLayer) error {
 	configFile := path.Join(minioConfigPrefix, minioConfigFile)
 	srvConfig, err := readServerConfig(context.Background(), objAPI)
 	if err == errConfigNotFound {
 		return nil
 	} else if err != nil {
 		return fmt.Errorf("Unable to load config file. %v", err)
 	}
-	if srvConfig.Version != "27" {
+	if srvConfig.Version != "28" {
 		return nil
 	}
 
-	srvConfig.Version = "28"
-	srvConfig.KMS = crypto.KMSConfig{}
+	srvConfig.Version = "29"
+	// Init compression config.For future migration, Compression config needs to be copied over from previous version.
+	srvConfig.Compression.Enabled = false
+	srvConfig.Compression.Extensions = []string{".txt", ".log", ".csv", ".json"}
```

Comment:
Can we have a global value for the defaults which can be added in future?

Reply:
Done